indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -388,16 +792,298 @@ public void shutdown(final String taskid)
+            }
+        );
+    if (taskDirFileList.isEmpty()) {
+      log.info("No task dirs found in [%s]", baseDir);
+      return;
+    }
+
+    // For the task directories, look for attempt directories
+    for (File potentialTaskDir : taskDirFileList) {
+      final File[] taskAttemptDirFileArray = potentialTaskDir.listFiles();
+      final Collection<File> taskAttemptDirFileList = // Only directories containing a non-zero TASK_FILE_NAME file
+          taskAttemptDirFileArray == null ?
+          ImmutableList.<File>of() :
+          Collections2.filter(
+              Arrays.asList(taskAttemptDirFileArray),
+              new Predicate<File>()
+              {
+                @Override
+                public boolean apply(File input)
+                {
+                  return input.isDirectory() && input.listFiles(
+                      new FilenameFilter()
+                      {
+                        @Override
+                        public boolean accept(File dir, String name)
+                        {
+                          return TASK_FILE_NAME.equals(name)
+                                 && new File(dir, TASK_FILE_NAME).length()
+                                    > 0;
+                        }
+                      }
+                  ) != null; // Did we find any?
+                }
+              }
+          );
+      if (taskAttemptDirFileList.isEmpty()) {
+        log.info("Directory [%s] has no viable task attempts, attempting to cleanup if empty", potentialTaskDir);
+        if (!potentialTaskDir.delete()) {
+          log.warn("Could not clean up [%s]", potentialTaskDir);
+        }
+        continue;
+      }
+      // Find latest attempt in directory
+      long last_attempt = 0;
+      File latestAttemptDir = null;
+      for (File taskAttemptDir : taskAttemptDirFileList) {
+        if (!taskAttemptDir.isDirectory()) {
+          log.debug("Skipping non-directory [%s]", taskAttemptDir);
+          continue;
+        }
+        try {
+          final long check_attempt = Long.parseLong(taskAttemptDir.getName());
+          if (check_attempt > last_attempt) {
+            latestAttemptDir = getTaskAttemptDir(taskAttemptDir.getName(), check_attempt);
+            last_attempt = check_attempt;
+          }
+        }
+        catch (NumberFormatException e) {
+          log.debug(e, "Skipping unparsable directory [%s]", taskAttemptDir);
+        }
+        if (latestAttemptDir == null) {
+          latestAttemptDir = taskAttemptDir;
+        } else {
+          if (latestAttemptDir.lastModified() < taskAttemptDir.lastModified()) {
+            latestAttemptDir = taskAttemptDir;
+          }
+        }
+      }
+      if (latestAttemptDir == null) {
+        log.wtf("I had directories in [%s] but now I don't... memory corruption?", potentialTaskDir);
+        continue;
+      }
+
+      //------------------------------------------------------------------ Load up data from suspected good attempt dirs
```

Comment:
A comment like this is a sign that this method should be split up into multiple methods.